doc/level1.rst

```diff
@@ -669,24 +687,28 @@ Updating Autocrypt Peer State from Key Gossip
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 An incoming message may contain one or more ``Autocrypt-Gossip``
-headers in the encrypted payload. Each of these headers may update the
-Autocrypt peer state of the gossiped recipient identified by its
-``addr`` value (referred to here as ``gossip-addr``) in the following
-way:
+headers in the encrypted payload. Each of these headers MUST
+contain the ``keydata``, ``addr`` and ``origin-timestamp`` attributes
+as specified in `gossip-injection`_.
 
-1. If ``gossip-addr`` does not match any recipient in the mail's
-   ``To`` or ``Cc`` header, the update process terminates (i.e.,
-   header is ignored).
+Update the Autocrypt gossip state for the recipient identified
+by the ``addr`` value (referenced as ``gossip-addr`` below) if
+and only if all of the following conditions are true:
 
-2. If ``peers[gossip-addr].gossip_timestamp`` is more recent than the
-   message's effective date, then the update process terminates.
+- ``gossip-addr`` matches a recipient in the mail's ``To`` or ``Cc``
+  header,
 
-3. Set ``peers[gossip-addr].gossip_timestamp`` to the message's
-   effective date.
+- ``origin-timestamp`` is less recent that the current date,
```

Comment:
should be "less recent than", or perhaps just "not in the future" -- or should it be the message's effective date?